File: OldProject/SpaceFist/SpaceFist/Weapons/Bluelaser.cs
using SpaceFist.Managers;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SpaceFist.Weapons
{
    /// <summary>
    /// Represents a weapon which drops stationary mines when fired.
    /// </summary>
    class Bluelaser : Weapon
    {
        private ProjectileManager projectileManager;
        private GameData gameData;

        /// <summary>
        /// Creates a new Bluelaser instance.
        /// </summary>
        /// <param name="gameData">Common game data</param>
        /// <param name="ship">The players ship</param>
        public Bluelaser(GameData gameData)
        {
            this.gameData = gameData;

            projectileManager = gameData.ProjectileManager;
        }

        /// <summary>
        /// Drops a mine at the ships location.
        /// </summary>
        public void fire()
        {
            Ship ship = gameData.Ship;

            int projectileX = (int)(ship.X + (ship.Rectangle.Width / 2) - (20 * gameData.ScreenScale));
            int projectileY = (int)((ship.Y - (35 * gameData.ScreenScale)));

            projectileManager.fireBluelaser(projectileX, projectileY);
        }
    }
}

File: core/src/com/spacefist/weapons/Bluelaser.java
package com.spacefist.weapons;

import com.spacefist.GameData;
import com.spacefist.entities.Ship;
import com.spacefist.managers.ProjectileManager;
import com.spacefist.weapons.abst.Weapon;
import org.jetbrains.annotations.NotNull;

/**
 * Represents a weapon which drops stationary mines when fired.
 */
public class Bluelaser implements Weapon {
    private ProjectileManager projectileManager;
    private GameData gameData;

    /**
     * Creates a new Bluelaser instance.
     *
     * @param gameData Common game data
     */
    public Bluelaser(@NotNull GameData gameData) {
        this.gameData = gameData;

        projectileManager = gameData.getProjectileManager();
    }

    /**
     * Drops a mine at the ships location.
     */
    @Override
    public void fire() {
        Ship ship = gameData.getShip();

        float shipWidth = ship.getRectangle().getWidth();
        float shipHeight = ship.getRectangle().getHeight();


        int projectileX = (int) ((ship.getX() + (shipWidth / 2)) - (20 * gameData.getScreenScale()));
        int projectileY = (int) (ship.getY() + (shipHeight * 1.125f));

        projectileManager.fireBluelaser(projectileX, projectileY);
    }
}
